stl/src/fiopen.cpp

```diff
@@ -58,13 +57,12 @@ FILE* _Xfiopen(const CharT* filename, ios_base::openmode mode, int prot) {
 
     mode &= ~(ios_base::ate | ios_base::_Nocreate | ios_base::_Noreplace);
 
+    // look for a valid mode
     int n = 0;
-    while (valid[n] != 0 && valid[n] != mode) { // look for a valid mode
-        ++n;
-    }
-
-    if (valid[n] == 0) {
-        return nullptr; // no valid mode
+    for (; valid[n] != mode; ++n) {
+        if (n == size(valid)) {
+            return nullptr; // no valid mode
+        }
     }

```

Comment:
We need to terminate the loop _before_ examining `valid[size(valid)]`, which is past-the-end of the array:
```suggestion
    while (valid[n] != mode) {
        if (++n == _STD size(valid)) {
            return nullptr; // no valid mode
        }
    }
```